infraci.jenkins.io-agents-2.tf

```
@@ -0,0 +1,162 @@
+  virtual_network_name = data.azurerm_virtual_network.infra_ci_jenkins_io.name
+}
+
+#trivy:ignore:avd-azu-0040 # No need to enable oms_agent for Azure monitoring as we already have datadog
+resource "azurerm_kubernetes_cluster" "infracijenkinsio_agents_2" {
+  name     = local.aks_clusters["infracijenkinsio_agents_2"].name
+  sku_tier = "Standard"
+  ## Private cluster requires network setup to allow API access from:
+  # - infra.ci.jenkins.io agents (for both terraform job agents and kubernetes-management agents)
+  # - private.vpn.jenkins.io to allow admin management (either Azure UI or kube tools from admin machines)
+  private_cluster_enabled             = true
+  private_cluster_public_fqdn_enabled = true
+  dns_prefix                          = "infracijenkinsioagents2" # Avoid hyphens in this DNS host
+  location                            = azurerm_resource_group.infracijio_kubernetes_agents.location
+  resource_group_name                 = azurerm_resource_group.infracijio_kubernetes_agents.name
+  kubernetes_version                  = local.aks_clusters["infracijenkinsio_agents_2"].kubernetes_version
+  role_based_access_control_enabled   = true # default value but made explicit to please trivy
+
+  upgrade_override {
+    # TODO: disable to avoid "surprise" upgrades
+    force_upgrade_enabled = true
+  }
+
+  image_cleaner_interval_hours = 48
+
+  network_profile {
+    network_plugin      = "azure"
+    network_plugin_mode = "overlay"
+    network_policy      = "azure"
+    outbound_type       = "userAssignedNATGateway"
+    load_balancer_sku   = "standard" # Required to customize the outbound type
+    pod_cidr            = local.aks_clusters.infracijenkinsio_agents_2.pod_cidr
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  default_node_pool {
+    name                         = "systempool1"
+    only_critical_addons_enabled = true # This property is the only valid way to add the "CriticalAddonsOnly=true:NoSchedule" taint to the default node pool
+    vm_size                      = "Standard_D2pds_v5"
+    temporary_name_for_rotation  = "syspooltemp"
+    upgrade_settings {
+      max_surge = "10%"
+    }
+    os_sku               = "AzureLinux"
+    os_disk_type         = "Ephemeral"
+    os_disk_size_gb      = 75 # Ref. Cache storage size at https://learn.microsoft.com/fr-fr/azure/virtual-machines/dasv5-dadsv5-series#dadsv5-series (depends on the instance size)
+    orchestrator_version = local.aks_clusters["infracijenkinsio_agents_2"].kubernetes_version
+    kubelet_disk_type    = "OS"
+    auto_scaling_enabled = true
+    min_count            = 2 # for best practices
+    max_count            = 3 # for upgrade
+    vnet_subnet_id       = data.azurerm_subnet.infraci_jenkins_io_kubernetes_agent.id
+    tags                 = local.default_tags
+    # Avoid deploying system pool in the same zone as other node pools
+    zones = [for zone in local.aks_clusters.infracijenkinsio_agents_2.compute_zones : zone + 1]
```

```suggestion
    # Avoid deploying system pool in the same zone as other node pools
    zones = [for zone in local.aks_clusters.infracijenkinsio_agents_2.compute_zones : zone + 1]
```

Pair review: let's specify multiple zones to decrease risks of instance not available when upgrading. See https://github.com/jenkins-infra/azure/blob/525247b99ad7a66cfe84fb234b8bab751db1a060/privatek8s-sponsorship.tf#L60C1-L60C155